jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererUtil.kt

```diff
@@ -44,8 +44,12 @@ import software.aws.toolkits.telemetry.CodewhispererCompletionType
 import software.aws.toolkits.telemetry.CodewhispererGettingStartedTask
 
 fun <T> calculateIfIamIdentityCenterConnection(project: Project, calculationTask: (connection: ToolkitConnection) -> T): T? =
-    ToolkitConnectionManager.getInstance(project).activeConnectionForFeature(CodeWhispererConnection.getInstance())?.let {
-        calculateIfIamIdentityCenterConnection(it, calculationTask)
+    if (project.isDisposed) {
+        null
+    } else {
+        ToolkitConnectionManager.getInstance(project).activeConnectionForFeature(CodeWhispererConnection.getInstance())?.let {
+            calculateIfIamIdentityCenterConnection(it, calculationTask)
+        }
```

Comment:
I don't think we will need this, the above check in `activeConnectionChanged` should be good. I prefer not to hide too many issues if there are at the same time.

Reply:
sg